vpr/src/draw/draw_types.h

```diff
@@ -141,86 +143,132 @@ typedef struct {
     e_draw_routing_costs show_routing_costs;
+
+    ///@brief Toggles whether block pin util is shown
     e_draw_block_pin_util show_blk_pin_util = DRAW_NO_BLOCK_PIN_UTIL;
+
+    ///@brief Toggles whether router expansion cost is shown
     e_draw_router_expansion_cost show_router_expansion_cost = DRAW_NO_ROUTER_EXPANSION_COST;
+
+    ///@brief Toggles whether placement macros are shown
     e_draw_placement_macros show_placement_macros = DRAW_NO_PLACEMENT_MACROS;
+
     int show_routing_bb = OPEN;
+
+    ///@brief toggles whether routing util is shown
     e_draw_routing_util show_routing_util = DRAW_NO_ROUTING_UTIL;
+
+    ///@brief Controls drawing of routing resources on screen, if pic_on_screen is ROUTING.
     e_draw_rr_toggle draw_rr_toggle = DRAW_NO_RR;
+
+    ///@brief Whether routing util is shown
     bool clip_routing_util = false;
+
+    ///@brief Boolean that toggles block outlines are shown
     bool draw_block_outlines = true;
+
+    ///@brief Boolean that toggles block names
     bool draw_block_text = true;
+
+    ///@brief integer value for net max fanout
     int draw_net_max_fanout = std::numeric_limits<int>::max();
+
+    ///@brief The maximum number of sub-block levels among all physical block types in the FPGA.
     int max_sub_blk_lvl = 0;
+
+    ///@brief If 0, no internal drawing is shown. Otherwise, indicates how many levels of sub-pbs to be drawn
     int show_blk_internal = 0;
+
+    ///@brief Whether graphics are enabled
     bool show_graphics = false;
+
+    ///@brief How often is user input required. (0: each t, 1: each place, 2: never)
     int gr_automode = 0;
+
+    ///@brief Should we automatically finish drawing (instead of waiting in the event loop for user interaction?
     bool auto_proceed = false;
+
+    ///@brief GLOBAL or DETAILED
     e_route_type draw_route_type = GLOBAL;
+
+    ///@brief default screen message on screen
     char default_message[vtr::bufsize];
+
+    ///@brief color in which each net should be drawn. [0..cluster_ctx.clb_nlist.nets().size()-1]
     vtr::vector<ClusterNetId, ezgl::color> net_color;
+
+    /**
+     * @brief stores the state information of each routing resource.
+     * 
+     * Used to control drawing each routing resource when
+     * ROUTING is on screen.
+     * [0..device_ctx.rr_nodes.size()-1]
+     */
     std::vector<t_draw_rr_node> draw_rr_node;
+
     std::shared_ptr<const SetupTimingInfo> setup_timing_info;
+
+    ///@brief pointer to architecture info. const
     const t_arch* arch_info = nullptr;
+
     std::shared_ptr<const vtr::ColorMap> color_map = nullptr;
+
+    ///@brief Whether to generate an output graphics file
     bool save_graphics = false;
+
     std::string graphics_commands;
+
+    ///@brief If we should pause for user interaction (requested by user)
     bool forced_pause = false;
+
     int sequence_number = 0;
     float net_alpha = 0.1;
+
+    ///@brief Present congestion cost factor
```

Comment:
Should expand this comment a bit. This is the pres_fac used to draw congestion costs; I believe it will be a copy of the router's current pres_fac, and is copied because the router's pres_fac is a local variable not accessible to drawing.

(you can check this with a quick code search).